### PR TITLE
fix!: convert R list to Series correctly via `as_polars_series()`

### LINKED ISSUE #1020: rewrite `as_polars_series()`
Related to #1012

Currently, due to the following processing, the types in list are processed ignoring the S3 method of `as_polars_series`, so appropriate type conversion is not performed.
https://github.com/pola-rs/r-polars/blob/3a08926db2a85da8fc0674741c55fc8717282bba/src/rust/src/conversion_r_to_s.rs#L177-L193

Like this:

``` r
polars::as_polars_series(list(data.frame(a = 1), data.frame(a = 2)))
#> polars Series: shape: (2,)
#> Series: '' [list[list[f64]]]
#> [
#>  [[1.0]]
#>  [[2.0]]
#> ]
arrow::as_arrow_array(list(data.frame(a = 1), data.frame(a = 2)))
#> ListArray
#> <list<item: struct<a: double>>>
#> [
#>   -- is_valid: all not null
#>   -- child 0 type: double
#>     [
#>       1
#>     ],
#>   -- is_valid: all not null
#>   -- child 0 type: double
#>     [
#>       2
#>     ]
#> ]
polars::as_polars_series(arrow::as_arrow_array(list(data.frame(a = 1), data.frame(a = 2))))
#> polars Series: shape: (2,)
#> Series: '' [list[struct[1]]]
#> [
#>  [{1.0}]
#>  [{2.0}]
#> ]
```

<sup>Created on 2024-04-09 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by eitsupi**
Fix #1020

Ideally, this would require significant refactoring on the Rust side, but for now it will work correctly.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -13,6 +13,37 @@
 
 ### Other breaking changes
 
+- R objects inside an R list are now converted to Polars data types via
+  `as_polars_series()` (#1021). For example, data.frames inside a list
+  are converted to the polars List type in previous versions.
```
Comment (etiennebacher) [inline NEWS.md:18]:
```suggestion
  `as_polars_series()` (#1021). For example, up to polars 0.15.1, 
  data.frames inside a list were converted to a nested List type:
```

File: NEWS.md [hunk 2]
```diff
@@ -13,6 +13,37 @@
...
+  #> │ nested_data     │
+  #> │ ---             │
+  #> │ list[list[f64]] │
+  #> ╞═════════════════╡
+  #> │ [[1.0]]         │
+  #> └─────────────────┘
+  ```
+
+  In 0.17.0, data.frames inside a list are converted to the polars Struct type
+  via `as_polars_series(<data.frame>)`.
```
Comment (etiennebacher) [inline NEWS.md:33]:
```suggestion
  From 0.16.0, data.frames inside a list are converted to the polars Struct type:
```

File: tests/testthat/test-as_polars.R [hunk 3]
```diff
@@ -427,3 +427,16 @@ test_that("clock_zoned_time may returns empty time zone", {
...
   expect_s3_class(as_polars_series(clock::zoned_time_now(zone = "")), "RPolarsSeries")
 })
+
+
+test_that("as_polars_series for nested type", {
+  expect_true(
+    as_polars_series(list(list(data.frame(a = 1))))$dtype == pl$List(pl$List(pl$Struct(a = pl$Float64)))
+  )
+
+  # TODO: Fix me
```
Comment (etiennebacher) [inline tests/testthat/test-as_polars.R:437]:
```suggestion
  # TODO: this shouldn't error
```
I guess?

**Review by etiennebacher — APPROVED** [review]

File: NEWS.md [hunk 4]
```diff
@@ -13,6 +13,36 @@
 
 ### Other breaking changes
 
+- R objects inside an R list are now converted to Polars data types via
+  `as_polars_series()` (#1021). For example, up to polars 0.15.1,
```
Comment (eitsupi) [inline NEWS.md:17]:
```suggestion
  `as_polars_series()` (#1021). For example, up to polars 0.15.1,
```

**Pull request merged**